src/rust/lib/system/web/src/event.rs

```diff
@@ -0,0 +1,42 @@
+//! Utilities for DOM events.
+
+pub mod listener;
+
+use crate::prelude::*;
+
+use js_sys::Function;
+use web_sys::EventTarget;
+
+
+
+// =============
+// === Event ===
+// =============
+
+/// This trait represents a type of event that may fire from some specific JS `EventTarget`.
+///
+/// For example, `WebSocket.close` is such an event, where `close` is event type and `WebSocket` is
+/// the `EventTarget`.
```

Comment:
I'm not sure if I get the docs correctly: so the trait is meant to be implemented on structures defined in js_sys?